File: VB/SyntaxHighlight/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit.Services.Implementation
Imports DevExpress.Services
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native

Namespace SyntaxHighlight
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			richEditControl1.LoadDocument("FairyTales.rtf", DocumentFormat.Rtf)

			Dim service As ISyntaxHighlightService = richEditControl1.GetService(Of ISyntaxHighlightService)()
			Dim wrapper As New MySyntaxHighlightServiceWrapper(richEditControl1)
			richEditControl1.RemoveService(GetType(ISyntaxHighlightService))
			richEditControl1.AddService(GetType(ISyntaxHighlightService), wrapper)
		End Sub

		Private Class MySyntaxHighlightServiceWrapper
			Implements ISyntaxHighlightService
			Private control As RichEditControl
			Private Shared str() As String = { "luck", "Hans" }
			Private direction As SearchDirection = SearchDirection.Forward
			Private options As SearchOptions = SearchOptions.WholeWord Or SearchOptions.CaseSensitive
			Public Sub New(ByVal control As RichEditControl)
				Me.control = control
			End Sub
			#Region "ISyntaxHighlightService Members"

			Public Sub Execute() Implements ISyntaxHighlightService.Execute
				Dim doc As Document = Me.control.Document
				For Each word As String In str
					Dim result As ISearchResult = control.Document.StartSearch(word, options, direction, doc.Range)
					Do While result.FindNext()
						Dim charProperties As CharacterProperties = doc.BeginUpdateCharacters(result.CurrentResult)
						charProperties.ForeColor = Color.IndianRed
						doc.EndUpdateCharacters(charProperties)
					Loop
				Next word
				' your code here                
			End Sub

			#End Region
		End Class
	End Class
End Namespace

File: CS/SyntaxHighlight/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraRichEdit.Services.Implementation;
using DevExpress.Services;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;

namespace SyntaxHighlight
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            richEditControl1.LoadDocument("FairyTales.rtf", DocumentFormat.Rtf);

            ISyntaxHighlightService service = richEditControl1.GetService<ISyntaxHighlightService>();
            MySyntaxHighlightServiceWrapper wrapper = new MySyntaxHighlightServiceWrapper(richEditControl1);
            richEditControl1.RemoveService(typeof(ISyntaxHighlightService));
            richEditControl1.AddService(typeof(ISyntaxHighlightService), wrapper);
        }

        class MySyntaxHighlightServiceWrapper : ISyntaxHighlightService
        {
            RichEditControl control;
            static string[] str = { "luck", "Hans" };
            SearchDirection direction = SearchDirection.Forward;
            SearchOptions options = SearchOptions.WholeWord |SearchOptions.CaseSensitive;
            public MySyntaxHighlightServiceWrapper(RichEditControl control) {
                this.control = control;
            }
            #region ISyntaxHighlightService Members

            public void Execute()
            {
                Document doc = this.control.Document;
                foreach (string word in str) {
                    ISearchResult result = control.Document.StartSearch(word, options, direction, doc.Range);
                    while (result.FindNext())
                    {
                        CharacterProperties charProperties = doc.BeginUpdateCharacters(result.CurrentResult);                        
                        charProperties.ForeColor = Color.IndianRed;
                        doc.EndUpdateCharacters(charProperties);
                    }
                }
                // your code here                
            }

            #endregion
        }
}
}